src/waterfall/waterfallplot.cpp

```diff
@@ -50,20 +50,13 @@ void WaterfallPlot::paint(QPainter* painter)
         _painter = painter;
     }
 
-    static uint16_t first;
-
-    if (_currentDrawIndex < _displayWidth) {
-        first = 0;
-    } else {
-        first = _currentDrawIndex - _displayWidth;
-    }
-
     // http://blog.qt.io/blog/2006/05/13/fast-transformed-pixmapimage-drawing/
     pix = QPixmap::fromImage(_image, Qt::NoFormatConversion);
     // Code for debug, draw the entire waterfall
-    //_painter->drawPixmap(_painter->viewport(), pix, QRect(0, 0, _image.width(), _image.height()));
-    _painter->drawPixmap(QRect(0, 0, width(), height()), pix,
-        QRect(first, _minDepthToDrawInPixels, _displayWidth, _maxDepthToDrawInPixels));
+    _painter->drawPixmap(_painter->viewport(), pix, QRect(0, 0, _image.width(), _image.height()));
+
+    emit drawHorizontalRatioChanged();
+    emit drawVerticalRatioChanged();
```

Comment:
what do you think about merging those two emits into one? if the horizontal and vertical changes at the same time *probably* there will be one less redraw